tests/test_exporters_jaeger.py

```diff
@@ -38,57 +38,57 @@ def setUp(self):
         self._test_span.start()
         self._test_span.end()
 
-        self.connection_patcher = mock.patch("http.client.HTTPConnection")
+        self.connection_patcher = patch("http.client.HTTPConnection")
         self.connection_mock = self.connection_patcher.start()
         conn = self.connection_mock.return_value
-        response = http.client.HTTPResponse(mock.Mock())
+        response = http.client.HTTPResponse(Mock())
         response.status = 200
         conn.getresponse.return_value = response
 
     def tearDown(self):
         self.connection_patcher.stop()
 
+    @patch.dict(os.environ, {"OTEL_TRACES_EXPORTER": "jaeger-thrift-splunk"})
     def test_exporter_uses_collector_not_udp_agent(self):
-        exporter = _new_jaeger_exporter(
-            Options(
-                endpoint=self.endpoint,
-                resource_attributes={"service.name": self.service_name},
-            )
-        )
-        agent_client_mock = mock.Mock(spec=jaeger_exporter.AgentClientUDP)
+        options = Options()
+        exporter = options.span_exporter_factories[0](options)
+        agent_client_mock = Mock(spec=jaeger_exporter.AgentClientUDP)
         exporter._agent_client = agent_client_mock  # pylint:disable=protected-access
-        collector_mock = mock.Mock(spec=jaeger_exporter.Collector)
+        collector_mock = Mock(spec=jaeger_exporter.Collector)
         exporter._collector = collector_mock  # pylint:disable=protected-access
 
         exporter.export((self._test_span,))
         self.assertEqual(agent_client_mock.emit.call_count, 0)
         self.assertEqual(collector_mock.submit.call_count, 1)
 
+    @patch.dict(
+        os.environ,
+        {
+            "OTEL_TRACES_EXPORTER": "jaeger-thrift-splunk",
+        },
+    )
     def test_http_export(self):
-        exporter = _new_jaeger_exporter(
-            Options(
-                endpoint=self.endpoint,
-                resource_attributes={"service.name": self.service_name},
-            )
-        )
+        options = Options()
+        exporter = options.span_exporter_factories[0](options)
         exporter.export((self._test_span,))
 
         conn = self.connection_mock.return_value
         conn.putrequest.assert_called_once_with("POST", "/v1/trace")
         conn.putheader.assert_any_call("User-Agent", "Python/THttpClient (pytest)")
```

Comment:
this assertion occurs to be too strict.
I propose to remove this line completely (I do not see any value TBH) or change to
```
from unittest.mock import ANY, Mock, patch
...
        conn.putheader.assert_any_call("User-Agent", "Python/THttpClient (pytest)")
```

Reply:
removed as per slack discussion.